parquet-column/src/main/java/org/apache/parquet/column/values/dictionary/IntList.java

```diff
@@ -31,7 +31,13 @@
  */
 public class IntList {
 
-  private static final int SLAB_SIZE = 64 * 1024;
+  public static final int MAX_SLAB_SIZE = 64 * 1024;
+  public static final int INITIAL_SLAB_SIZE = 4 * 1024;
+
+  //Double slab size till we reach the max slab size. At that point we just add slabs of size
+  //MAX_SLAB_SIZE. This ensures we don't allocate very large slabs from the start if we don't have
+  //too much data.
+  private int currentSlabSize = INITIAL_SLAB_SIZE / 2;
```

Comment:
Yeah so if I did:
```
private void initSlab() {
    currentSlab = new int[currentSlabSize];
    currentSlabPos = 0;
    updateCurrentSlabSize();
  }
```

That will mean that currentSlabSize at the end of initSlab points to the next slabSize that we want to use (not the current one in use, which I don't think is ideal). I could break up the initSlab to do just the array allocation in one method and the update slab size in a different method. Something like this:
```
private void initSlab() {
    currentSlab = new int[currentSlabSize];
    currentSlabPos = 0;
  }
  // Double slab size up to the MAX_SLAB_SIZE limit
  private void updateCurrentSlabSize() {
    if (currentSlabSize < MAX_SLAB_SIZE) {
      currentSlabSize *= 2;
      if (currentSlabSize > MAX_SLAB_SIZE) {
        currentSlabSize = MAX_SLAB_SIZE;
      }
    }
  }
```
Downstream where we are calling initSlab I can also make a call to updateCurrentSlabSize. 
e.g. 
```
if (slabIsFull) {
 updateCurrentSlabSize()
 initSlab()
}
```
Does this sound reasonable? Or do you have something else in mind?